static/app/views/performance/traceDetails/utils.tsx

```diff
@@ -68,12 +79,28 @@ export function getTraceDetailsUrl({
     queryParams.limit = DEFAULT_TRACE_ROWS_LIMIT;
   }
 
-  return {
-    pathname: normalizeUrl(
-      `/organizations/${organization.slug}/performance/trace/${traceSlug}/`
-    ),
-    query: queryParams,
-  };
+  return oldTraceUrl;
+}
+
+/**
+ * Single tenant, on-premise etc. users may not have span extraction enabled.
+ *
+ * This code can be removed at the time we're sure all STs have rolled out span extraction.
+ */
+export function shouldForceRouteToOldView(
+  organization: Organization,
+  timestamp: string | number | undefined
+) {
+  const usableTimestamp = getTimeStampFromTableDateField(timestamp);
+  if (!usableTimestamp) {
+    // Timestamps must always be provided for the new view, if it doesn't exist, fall back to the old view.
```

Comment:
@k-fish , it just helps make the query faster. The traceview still loads without it.

Reply:
I realize it works on the api but the frontend it should be mandatory no? it significantly improves trace performance, I had assumed all callsites were sending it where appropriate for the new view.

Reply:
Yeah we do it from all callsites. However, even without it the new traceview should load faster than the old traceview I believe. But yeah, just wanted to point it out in case you thought that the new view wouldn't work at all without timestamp.